packages/cursorless-engine/src/languages/LanguageDefinition.ts

```diff
@@ -91,12 +88,12 @@ export class LanguageDefinition {
  * @returns The text of the query file, with all imports inlined
  */
 async function readQueryFileAndImports(
-  fileSystem: FileSystem,
-  languageQueryPath: string,
+  provider: LanguageDefinitionsProvider,
+  languageFilename: string,
```

Comment:
The reason for switching from path to name is that if we have a non file system implementation of the language provider path is irrelevant. I think every definition name should just be unique and in which folder they are located doesn't really matter.